### ISSUE TITLE
.NET Native: LINQ query with anonymous types throw NullReferenceException

### ISSUE THREAD

**Issue by natemcmaster** [dotnet/efcore#5119]
With UWP Tooling 1.3.1 (latest at time of writing), some queries that use anonymous types fail on .NET Native (aka Universal Windows Platform apps on "Release" mode).

Example query that fails:

``` c#
from e in es.Select(e => new { e })
from o in os.Select(o => new { o })
where e.e.EmployeeID == o.o.EmployeeID
select new { e, o }
```

Example error:

```
System.NullReferenceException: Arg_NullReferenceException
   at UserProject!<BaseAddress>+0xa0b22c
   at SharedLibrary!<BaseAddress>+0x37975c
   at Microsoft.EntityFrameworkCore.Query.Internal.LinqOperatorProvider.ExceptionInterceptor<>.EnumeratorExceptionInterceptor.MoveNext()
   at SharedLibrary!<BaseAddress>+0x2dca0f
```
## Workaround

Replace anonymous types with concrete types. Although a little cumbersome, this should fix this issue. It also improves the speed _and_ functionality of .NET Native's LINQ interpretation.

**Comment by divega** [comment]
Curious about what the failure looks like.

**Comment by natemcmaster** [comment]
Added example error and stack trace.

**Comment by divega** [comment]
Thanks. Failure isn't great. I hope this can be fixed in .NET Native. I didn't know that nominal types made LINQ interpretation faster. That is interesting.

**Comment by rowanmiller** [comment]
We are discussing this with the .NET Native team, hopefully they can do something to enable this for us. Leaving open so that we can track the scenario (and marking as blocked since the fix is external to us).

**Comment by GHBI** [comment]
I did not use anonymous types, but in the debug is very fast, less than half a second. In relese very slow, there are more than ten seconds.
await db.Bibles.Where (m => m.Scripture.Contains (text)) ToListAsync ().;

**Comment by divega** [comment]
@GHBI this issue is specifically about a failure with anonymous types. You should be able to find other .NET Native issues that are performance related if you do some searching. If you aren't sure whether what you are seeing has already been reported please create a new issue and provide a repro.

**Comment by AndriySvyryd** [comment]
Verified that this is fixed in the prerelease builds of EF 2.0.1 and UWP 6.0.0
